javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/concurrent/ExecutorInstrumentationUtils.java

```diff
@@ -28,22 +65,40 @@ public static boolean shouldAttachStateToTask(Object task) {
     Class<?> taskClass = task.getClass();
     Class<?> enclosingClass = taskClass.getEnclosingClass();
 
-    // not much point in propagating root context
-    // plus it causes failures under otel.javaagent.testing.fail-on-context-leak=true
-    return Context.current() != Context.root()
-        // TODO Workaround for
-        // https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/787
-        && !taskClass.getName().equals("org.apache.tomcat.util.net.NioEndpoint$SocketProcessor")
-        // Avoid context leak on jetty. Runnable submitted from SelectChannelEndPoint is used to
-        // process a new request which should not have context from them current request.
-        && (enclosingClass == null
-            || !enclosingClass.getName().equals("org.eclipse.jetty.io.nio.SelectChannelEndPoint"))
-        // Don't instrument the executor's own runnables. These runnables may never return until
-        // netty shuts down.
-        && (enclosingClass == null
-            || !enclosingClass
-                .getName()
-                .equals("io.netty.util.concurrent.SingleThreadEventExecutor"));
+    if (Context.current() == Context.root()) {
+      // not much point in propagating root context
+      // plus it causes failures under otel.javaagent.testing.fail-on-context-leak=true
+      return false;
+    }
+
+    // TODO Workaround for
+    // https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/787
+    if (taskClass.getName().equals("org.apache.tomcat.util.net.NioEndpoint$SocketProcessor")) {
+      return false;
+    }
+
+    // Long running task which decorates other tasks. We should probably be using ExecutionContext
+    // for propagation in Scala but in the meantime this is not needed along with other concurrent
+    // instrumentation and causes context leaks.
+    if (taskClass.getName().equals("akka.util.SerializedSuspendableExecutionContext")) {
+      return false;
+    }
+
+    // Wrapper of tasks for dispatch - the wrapped task should have context already and this doesn't
```

Comment:
No idea what I'm talking about! But seems like a reasonable hypothesis for now, how akka flakes.